x/tokenfactory/keeper/msg_server.go

```diff
@@ -87,7 +91,11 @@ func (server msgServer) Burn(goCtx context.Context, msg *types.MsgBurn) (*types.
 		return nil, types.ErrUnauthorized
 	}
 
-	err = server.Keeper.burnFrom(ctx, msg.Amount, msg.Sender)
+	if msg.BurnFromAddress == "" {
+		msg.BurnFromAddress = msg.Sender
+	}
+
+	err = server.Keeper.burnFrom(ctx, msg.Amount, msg.BurnFromAddress)
```

Comment:
uf, are you sure that you want to let a contract burn their tokens from random accounts? This change should be discussed and not be hidden in this big PR. 
Don't get me wrong. I am not against this. There is a similar concept in Stellar and it makes sense for some use cases. But I am missing transparency and some safety mechanisms. 

Please make a proper proposal with a GH issue.

Reply:
This implements everything to the latest of osmosis, which added transferFrom, burnFrom, and mintTo features

To allow us to put this in, I am happy to add an internalCapabilities so each importing chain can specify the features they do not want on

Reply:
Osmosis has a different setup as a governanced chain. They don't run untrusted contracts.
For public chains, this can not be copied.

Reply:
@alpe what if I added capabilities like how wasmd does it, but for tfCapabilities := "force-transfer,burn-from" ?

Prob use enum types so chains can build features and upstream as needed, then other chains can implement on upgrades

Reply:
Using feature flags to enable certain functionalities is a good idea, generally speaking. 
But IMHO the design of `force-transfer`/ `burn-from` is completely broken on the product side for open chains. A user has no chance to trust the coins in his wallet to be permanent. Everything can be taken away by the contract (owner) without any notice. You can not find this a good idea. 🤷‍♂️